packages/runtime-common/index-writer.ts

```diff
@@ -250,6 +244,71 @@ export class Batch {
     return parseInt(total);
   }
 
+  private async updateRealmMeta() {
+    let results = await this.query([
+      `SELECT CAST(count(i.url) AS INTEGER) as total, i.display_names->>0 as display_name, i.types->>0 as code_ref
+       FROM boxel_index as i
+       INNER JOIN realm_versions r ON i.realm_url = r.realm_url
+          WHERE`,
+      ...every([
+        ['i.realm_url =', param(this.realmURL.href)],
+        ['i.type = ', param('instance')],
+        ['i.types IS NOT NULL'],
+        realmVersionExpression({
+          useWorkInProgressIndex: true,
+          withMaxVersion: this.realmVersion,
+        }),
+      ]),
+      `GROUP BY i.display_names->>0, i.types->>0`,
+    ] as Expression);
+
+    let { nameExpressions, valueExpressions } = asExpressions(
+      {
+        realm_url: this.realmURL.href,
+        realm_version: this.realmVersion,
+        value: results,
+        indexed_at: unixTime(new Date().getTime()),
+      } as Omit<RealmMetaTable, 'indexed_at'> & {
+        indexed_at: number;
+      },
+      {
+        jsonFields: ['value'],
+      },
+    );
+
+    await this.query([
+      ...upsert(
+        'realm_meta',
+        'realm_meta_pkey',
+        nameExpressions,
+        valueExpressions,
+      ),
+    ]);
+  }
+
+  private async applyBatchUpdates() {
+    let { nameExpressions, valueExpressions } = asExpressions({
+      realm_url: this.realmURL.href,
+      current_version: this.realmVersion,
+    } as RealmVersionsTable);
+    await this.query([
+      ...upsert(
+        'realm_versions',
+        'realm_versions_pkey',
+        nameExpressions,
+        valueExpressions,
+      ),
+    ]);
+    await this.query([
+      `DELETE FROM realm_meta`,
+      'WHERE',
+      ...every([
+        ['realm_version <', param(this.realmVersion)],
+        ['realm_url =', param(this.realmURL.href)],
+      ]),
+    ] as Expression);
+  }
```

Comment:
this is not very cohesive--the clean up is not part of making the batch updates live. keep in mind that we do clenup for the boxel_index as well after making the updates live, but you don't see that here. each method should be doing a single thing in order to have good cohesion. this method should be in charge of making the batch live. there should be a different method in chargetof cleanup after the batch is live.